### PR TITLE
Rename provider specific core metadata extensions

### LINKED ISSUE #16773: Remove provider prefix from low-level metadata APIs
In #16686 we removed all provider prefixes from builder extension methods, but the low-level metadata APIs have been left as-is. Discuss and decide whether we want to drop the prefix there as well.

### PR DESCRIPTION
**Pull request by ajcvickers**
Fixes #16773

Note going through obsolete here since this are a lot less used.

### CODE REVIEW

File: src/EFCore.Sqlite.Core/Extensions/SqlitePropertyExtensions.cs [hunk 1]
```diff
@@ -35,31 +35,31 @@ public static void SetSqliteSrid([NotNull] this IMutableProperty property, int?
...
+        public static ConfigurationSource? GetSridConfigurationSource([NotNull] this IConventionProperty property)
             => property.FindAnnotation(SqliteAnnotationNames.Srid)?.GetConfigurationSource();
 
         /// <summary>
         ///     Returns the dimension to use when creating a column for this property.
         /// </summary>
         /// <param name="property"> The property. </param>
         /// <returns> The dimension to use when creating a column for this property. </returns>
-        public static string GetSqliteDimension([NotNull] this IProperty property)
+        public static string GetSpatialDimension([NotNull] this IProperty property)
```
Comment (bricelam) [inline src/EFCore.Sqlite.Core/Extensions/SqlitePropertyExtensions.cs]: I have a slight preference for *geometry* dimension (builder APIs too)

Comment (bricelam) [reply src/EFCore.Sqlite.Core/Extensions/SqlitePropertyExtensions.cs]: (or possibly *geometric* if it grammars better)

Comment (ajcvickers) [reply src/EFCore.Sqlite.Core/Extensions/SqlitePropertyExtensions.cs]: How slight?

Comment (bricelam) [reply src/EFCore.Sqlite.Core/Extensions/SqlitePropertyExtensions.cs]: Significant. Neither NTS nor SpaiaLite use the word spatial in their APIs

**Pull request merged**